Sources/SwiftRefactor/IntegerLiteralUtilities.swift

```diff
@@ -17,7 +17,7 @@ import SwiftSyntax
 #endif
 
 extension IntegerLiteralExprSyntax {
-  public enum Radix {
+  public enum Radix: CaseIterable {
```

Comment:
As for the use-case in SourceKit-LSP, I'd just do:

```swift
for radix: IntegerLiteralExprSyntax.Radix in [.binary, .octal, .hex, .decimal] { ... }
```
This is clearer and more deterministic what it will do, compared to `.allCases`. 

I personally avoid `CaseIterable` unless there's a _strong_ reason to do. Users cannot control the order (maybe it might not matter in this specific use-case)

Reply:
Hmm, I'm doing this based on the discussion in https://github.com/apple/sourcekit-lsp/pull/1179#discussion_r1569570269. Code that is working with radix values via the APIs here can do so without having to reason about the particulars of the radix; it can handle all of them. That seems like it makes this a reasonable API to provide.

Reply:
Yes I know the discussion. But,

> it can handle all of them.

All of _known_ cases. Perhaps in the future, we might introduce a case that has "suffix". e.g. `0nA24nG_r62` (whatever it is). If we only handle known cases `[.binary, .octal, .hex, .decimal]`, the code would probably still work correctly because we'd not change the syntax of the current radix cases. But using `.allCases` can cause misbehavior.

FWIW https://github.com/apple/swift-syntax/pull/2292, we have a history of removing `CaseIterable`. I want to be really careful to re-adding it.

Reply:
I think it's unlikely to ever cause a problem, given that the literals have been stable for a decade, but neither is it more than an inconvenience if we don't have this. So I've removed it.